src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -6,8 +6,32 @@
 
 namespace Microsoft.Extensions.Primitives
 {
+    /// <summary>
+    /// An optimized representation of a substring.
+    /// </summary>
     public struct StringSegment : IEquatable<StringSegment>, IEquatable<string>
     {
+        /// <summary>
+        /// Initializes an instance of the <see cref="StringSegment"/> struct.
+        /// </summary>
+        /// <param name="buffer">
+        /// The <see cref="string"/>that acts as a buffer for this <see cref="StringSegment"/>.
+        /// In this case, the <see cref="StringSegment"/> represents the whole original <see cref="string"/>
+        /// or its absence in case the parameter is <code>null</code>.
```

Comment:
I'd do something like this. These summaries need to be relatively short. You really **can't** effectively convey lots of info here.

```
The original <see cref="string"/> used as buffer.  The <see cref="StringSegment"/> includes the whole <see cref="string"/>.
```
I'd leave out the part about null.